File: August-LeetCoding-Challenge/11-H-Index/H-Index.cs
public class Solution {
    public int HIndex(int[] citations) {
        //Counting sort
        int[] counts = new int[citations.Length + 2];
        for (int i = citations.Length - 1; i >= 0; i--)
        {
            var idx = Math.Min(citations[i], citations.Length);
            counts[idx]++;
        }

        var r = 0;
        for (int i = citations.Length +1; i >= 0; i--)
        {
            r += counts[i];
            if (r >= i)
                return i;
        }
        return 0;
    }
}

File: August-LeetCoding-Challenge/11-H-Index/H-Index.js
var hIndex = function(citations) {
    if(citations.length==0){return 0;}
    citations.sort(function(a,b){
        return b-a;
    });
    var res = 0;
    if(citations[0] < 1){return 0;}
    for(let i=0;i<citations.length;i++){
        let a = i+1;
        if(a==citations[i]){return a;}
        if(a > citations[i]){
            let reste = citations.length-a+1;
            res = citations[i];
            while(citations.length-res > reste){res++;}
            return res;
        }
    }
    return citations.length;
};
